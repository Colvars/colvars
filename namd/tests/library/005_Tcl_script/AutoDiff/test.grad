# 1
#  1.20000000000000e+01  2.00000000000000e-01        100  0

  1.21000000000000e+01   6.99836008077528e+01
  1.23000000000000e+01  -5.26375313396654e+02
  1.25000000000000e+01  -5.16708951944646e+02
  1.27000000000000e+01  -5.05977066222890e+02
  1.29000000000000e+01  -4.88119448094002e+02
  1.31000000000000e+01  -4.54995001210363e+02
  1.33000000000000e+01  -4.12314411188385e+02
  1.35000000000000e+01  -3.59715820941386e+02
  1.37000000000000e+01  -2.97680997044662e+02
  1.39000000000000e+01  -2.12126120236132e+02
  1.41000000000000e+01  -9.78647225049371e+01
  1.43000000000000e+01  -1.38644228524206e+01
  1.45000000000000e+01   0.00000000000000e+00
  1.47000000000000e+01   0.00000000000000e+00
  1.49000000000000e+01   0.00000000000000e+00
  1.51000000000000e+01   0.00000000000000e+00
  1.53000000000000e+01   0.00000000000000e+00
  1.55000000000000e+01   0.00000000000000e+00
  1.57000000000000e+01   0.00000000000000e+00
  1.59000000000000e+01   0.00000000000000e+00
  1.61000000000000e+01   0.00000000000000e+00
  1.63000000000000e+01   0.00000000000000e+00
  1.65000000000000e+01   0.00000000000000e+00
  1.67000000000000e+01   0.00000000000000e+00
  1.69000000000000e+01   0.00000000000000e+00
  1.71000000000000e+01   0.00000000000000e+00
  1.73000000000000e+01   0.00000000000000e+00
  1.75000000000000e+01   0.00000000000000e+00
  1.77000000000000e+01   0.00000000000000e+00
  1.79000000000000e+01   0.00000000000000e+00
  1.81000000000000e+01   0.00000000000000e+00
  1.83000000000000e+01   0.00000000000000e+00
  1.85000000000000e+01   0.00000000000000e+00
  1.87000000000000e+01   0.00000000000000e+00
  1.89000000000000e+01   0.00000000000000e+00
  1.91000000000000e+01   0.00000000000000e+00
  1.93000000000000e+01   0.00000000000000e+00
  1.95000000000000e+01   0.00000000000000e+00
  1.97000000000000e+01   0.00000000000000e+00
  1.99000000000000e+01   0.00000000000000e+00
  2.01000000000000e+01   0.00000000000000e+00
  2.03000000000000e+01   0.00000000000000e+00
  2.05000000000000e+01   0.00000000000000e+00
  2.07000000000000e+01   0.00000000000000e+00
  2.09000000000000e+01   0.00000000000000e+00
  2.11000000000000e+01   0.00000000000000e+00
  2.13000000000000e+01   0.00000000000000e+00
  2.15000000000000e+01   0.00000000000000e+00
  2.17000000000000e+01   0.00000000000000e+00
  2.19000000000000e+01   0.00000000000000e+00
  2.21000000000000e+01   0.00000000000000e+00
  2.23000000000000e+01   0.00000000000000e+00
  2.25000000000000e+01   0.00000000000000e+00
  2.27000000000000e+01   0.00000000000000e+00
  2.29000000000000e+01   0.00000000000000e+00
  2.31000000000000e+01   0.00000000000000e+00
  2.33000000000000e+01   0.00000000000000e+00
  2.35000000000000e+01   0.00000000000000e+00
  2.37000000000000e+01   0.00000000000000e+00
  2.39000000000000e+01   0.00000000000000e+00
  2.41000000000000e+01   0.00000000000000e+00
  2.43000000000000e+01   0.00000000000000e+00
  2.45000000000000e+01   0.00000000000000e+00
  2.47000000000000e+01   0.00000000000000e+00
  2.49000000000000e+01   0.00000000000000e+00
  2.51000000000000e+01   0.00000000000000e+00
  2.53000000000000e+01   0.00000000000000e+00
  2.55000000000000e+01   0.00000000000000e+00
  2.57000000000000e+01   0.00000000000000e+00
  2.59000000000000e+01   0.00000000000000e+00
  2.61000000000000e+01   0.00000000000000e+00
  2.63000000000000e+01   0.00000000000000e+00
  2.65000000000000e+01   0.00000000000000e+00
  2.67000000000000e+01   0.00000000000000e+00
  2.69000000000000e+01   0.00000000000000e+00
  2.71000000000000e+01   0.00000000000000e+00
  2.73000000000000e+01   0.00000000000000e+00
  2.75000000000000e+01   0.00000000000000e+00
  2.77000000000000e+01   0.00000000000000e+00
  2.79000000000000e+01   0.00000000000000e+00
  2.81000000000000e+01   0.00000000000000e+00
  2.83000000000000e+01   0.00000000000000e+00
  2.85000000000000e+01   0.00000000000000e+00
  2.87000000000000e+01   0.00000000000000e+00
  2.89000000000000e+01   0.00000000000000e+00
  2.91000000000000e+01   0.00000000000000e+00
  2.93000000000000e+01   0.00000000000000e+00
  2.95000000000000e+01   0.00000000000000e+00
  2.97000000000000e+01   0.00000000000000e+00
  2.99000000000000e+01   0.00000000000000e+00
  3.01000000000000e+01   0.00000000000000e+00
  3.03000000000000e+01   0.00000000000000e+00
  3.05000000000000e+01   0.00000000000000e+00
  3.07000000000000e+01   0.00000000000000e+00
  3.09000000000000e+01   0.00000000000000e+00
  3.11000000000000e+01   0.00000000000000e+00
  3.13000000000000e+01   0.00000000000000e+00
  3.15000000000000e+01   0.00000000000000e+00
  3.17000000000000e+01   0.00000000000000e+00
  3.19000000000000e+01   0.00000000000000e+00
