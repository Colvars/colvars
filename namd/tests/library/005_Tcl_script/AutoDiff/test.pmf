# 1
#  1.19000000000000e+01  2.00000000000000e-01        101  0

  1.20000000000000e+01   7.63151734965745e+02
  1.22000000000000e+01   7.77148455127296e+02
  1.24000000000000e+01   6.71873392447965e+02
  1.26000000000000e+01   5.68531602059036e+02
  1.28000000000000e+01   4.67336188814458e+02
  1.30000000000000e+01   3.69712299195657e+02
  1.32000000000000e+01   2.78713298953585e+02
  1.34000000000000e+01   1.96250416715908e+02
  1.36000000000000e+01   1.24307252527630e+02
  1.38000000000000e+01   6.47710531186980e+01
  1.40000000000000e+01   2.23458290714716e+01
  1.42000000000000e+01   2.77288457048417e+00
  1.44000000000000e+01   0.00000000000000e+00
  1.46000000000000e+01   0.00000000000000e+00
  1.48000000000000e+01   0.00000000000000e+00
  1.50000000000000e+01   0.00000000000000e+00
  1.52000000000000e+01   0.00000000000000e+00
  1.54000000000000e+01   0.00000000000000e+00
  1.56000000000000e+01   0.00000000000000e+00
  1.58000000000000e+01   0.00000000000000e+00
  1.60000000000000e+01   0.00000000000000e+00
  1.62000000000000e+01   0.00000000000000e+00
  1.64000000000000e+01   0.00000000000000e+00
  1.66000000000000e+01   0.00000000000000e+00
  1.68000000000000e+01   0.00000000000000e+00
  1.70000000000000e+01   0.00000000000000e+00
  1.72000000000000e+01   0.00000000000000e+00
  1.74000000000000e+01   0.00000000000000e+00
  1.76000000000000e+01   0.00000000000000e+00
  1.78000000000000e+01   0.00000000000000e+00
  1.80000000000000e+01   0.00000000000000e+00
  1.82000000000000e+01   0.00000000000000e+00
  1.84000000000000e+01   0.00000000000000e+00
  1.86000000000000e+01   0.00000000000000e+00
  1.88000000000000e+01   0.00000000000000e+00
  1.90000000000000e+01   0.00000000000000e+00
  1.92000000000000e+01   0.00000000000000e+00
  1.94000000000000e+01   0.00000000000000e+00
  1.96000000000000e+01   0.00000000000000e+00
  1.98000000000000e+01   0.00000000000000e+00
  2.00000000000000e+01   0.00000000000000e+00
  2.02000000000000e+01   0.00000000000000e+00
  2.04000000000000e+01   0.00000000000000e+00
  2.06000000000000e+01   0.00000000000000e+00
  2.08000000000000e+01   0.00000000000000e+00
  2.10000000000000e+01   0.00000000000000e+00
  2.12000000000000e+01   0.00000000000000e+00
  2.14000000000000e+01   0.00000000000000e+00
  2.16000000000000e+01   0.00000000000000e+00
  2.18000000000000e+01   0.00000000000000e+00
  2.20000000000000e+01   0.00000000000000e+00
  2.22000000000000e+01   0.00000000000000e+00
  2.24000000000000e+01   0.00000000000000e+00
  2.26000000000000e+01   0.00000000000000e+00
  2.28000000000000e+01   0.00000000000000e+00
  2.30000000000000e+01   0.00000000000000e+00
  2.32000000000000e+01   0.00000000000000e+00
  2.34000000000000e+01   0.00000000000000e+00
  2.36000000000000e+01   0.00000000000000e+00
  2.38000000000000e+01   0.00000000000000e+00
  2.40000000000000e+01   0.00000000000000e+00
  2.42000000000000e+01   0.00000000000000e+00
  2.44000000000000e+01   0.00000000000000e+00
  2.46000000000000e+01   0.00000000000000e+00
  2.48000000000000e+01   0.00000000000000e+00
  2.50000000000000e+01   0.00000000000000e+00
  2.52000000000000e+01   0.00000000000000e+00
  2.54000000000000e+01   0.00000000000000e+00
  2.56000000000000e+01   0.00000000000000e+00
  2.58000000000000e+01   0.00000000000000e+00
  2.60000000000000e+01   0.00000000000000e+00
  2.62000000000000e+01   0.00000000000000e+00
  2.64000000000000e+01   0.00000000000000e+00
  2.66000000000000e+01   0.00000000000000e+00
  2.68000000000000e+01   0.00000000000000e+00
  2.70000000000000e+01   0.00000000000000e+00
  2.72000000000000e+01   0.00000000000000e+00
  2.74000000000000e+01   0.00000000000000e+00
  2.76000000000000e+01   0.00000000000000e+00
  2.78000000000000e+01   0.00000000000000e+00
  2.80000000000000e+01   0.00000000000000e+00
  2.82000000000000e+01   0.00000000000000e+00
  2.84000000000000e+01   0.00000000000000e+00
  2.86000000000000e+01   0.00000000000000e+00
  2.88000000000000e+01   0.00000000000000e+00
  2.90000000000000e+01   0.00000000000000e+00
  2.92000000000000e+01   0.00000000000000e+00
  2.94000000000000e+01   0.00000000000000e+00
  2.96000000000000e+01   0.00000000000000e+00
  2.98000000000000e+01   0.00000000000000e+00
  3.00000000000000e+01   0.00000000000000e+00
  3.02000000000000e+01   0.00000000000000e+00
  3.04000000000000e+01   0.00000000000000e+00
  3.06000000000000e+01   0.00000000000000e+00
  3.08000000000000e+01   0.00000000000000e+00
  3.10000000000000e+01   0.00000000000000e+00
  3.12000000000000e+01   0.00000000000000e+00
  3.14000000000000e+01   0.00000000000000e+00
  3.16000000000000e+01   0.00000000000000e+00
  3.18000000000000e+01   0.00000000000000e+00
  3.20000000000000e+01   0.00000000000000e+00
