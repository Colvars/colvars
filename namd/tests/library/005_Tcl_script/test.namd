# -*- tcl -*-

# FILENAMES
set outName             [file rootname [file tail [info script]]]
#set inFileNum           [expr [scan [string range $outName end-1 end] "%d"] - 1]
#set inName              [format "%s%02u" [string range $outName 0 end-2] $inFileNum]

# FILES

#outputname              test01
#DCDfile                 test01.dcd

# MD SETUP
timestep          0.5
numSteps		  20

# Interactive molecular dynamics
#IMDon               on
#IMDport             3000
#IMDfreq             20
#IMDwait             on

colvars			  on
colvarsConfig	  test.in
#colvarsInput  $inName.colvars.state
#colvarsInput  test.colvars.state

#tclForces on
#tclForcesScript {
#  addatom 12
#  proc calcforces {} {
#    addforce 12 { 10 0 0 }
#  }
#}

# INPUT
structure               ../Common/da.psf 
#parameters              missing.par
parameters              ../Common/par_all22_prot.inp
paraTypeCharmm          on
coordinates             ../Common/da.min.pdb

temperature 250
#bincoordinates       $inName.coor
#binvelocities        $inName.vel

# OUTPUT

outputname              ${outName}
outputenergies          100
outputtiming            2000
binaryoutput            no
# binaryrestart           yes

#XSTFreq                 1000

# DCD TRAJECTORY

DCDfile                 ${outName}.dcd
DCDfreq                 2000

seed		87654321

# CONSTANT-T
langevin                on
langevinTemp		    300.0
langevinDamping         10.0
langevinHydrogen	    on

# SPACE PARTITIONING
splitpatch              hydrogen
hgroupcutoff            2.8
stepspercycle           20
margin                  1.0

# CUT-OFFS
switching               on
switchdist              9.0
cutoff                  11.0
pairlistdist            12.0

# RESPA 
fullElectFrequency      1
nonbondedFreq           1

# 1-4 NON-BONDED
exclude                 scaled1-4
1-4scaling              1.0

# COM
commotion               no

# SHAKE
rigidBonds              none

proc runcmd { cmd } {
  print "Running $cmd"
  print [eval $cmd]
}

foreach c [cv list] { print "$c = [cv colvar $c value]"}        
foreach b [cv list biases] { print "E($b) = [cv bias $b energy]"}

runcmd "cv update"

foreach c [cv list] { print "$c = [cv colvar $c value]"}        
foreach b [cv list biases] { print "E($b) = [cv bias $b energy]"}

run 20

foreach c [cv list] { print "$c = [cv colvar $c value]"}        
foreach b [cv list biases] { print "E($b) = [cv bias $b energy]"}

# cv config "colvar {
#   name d2
#   distance {
#     group1 { atomnumbers 4 }
#     group2 { atomnumbers 99 }
#   }
# }"
# runcmd "cv colvar d2 value"
# runcmd "cv update"
# runcmd "cv colvar d2 value"
# run 20
# runcmd "cv colvar d2 value"
# runcmd "cv colvar d2 delete"

cv config "harmonic {
  name harm2
  colvars d
  centers 13
  forceConstant 1.
  outputEnergy yes
}"
run 20 

runcmd "cv bias harmonic1 state"
runcmd "cv bias harmonic1 delete"

runcmd "cv bias harm2 energy"
runcmd "cv bias harm2 update"
run 20 
runcmd "cv bias harm2 energy"
runcmd "cv bias harm2 state"
runcmd "cv bias harm2 delete"

cv reset
# reinitialize just for the fun of it ;-)
cv configfile 	  test.in
