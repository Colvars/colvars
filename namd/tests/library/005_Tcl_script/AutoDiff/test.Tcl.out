TCL: Running Colvars version 2020-04-09
TCL: Running cv units real
TCL: 
TCL: d = 0.00000000000000e+00
TCL: E(abf1) = 0
TCL: E(harmonic1) = 0
TCL: d = 0.00000000000000e+00
TCL: Running cv update
TCL: 
TCL: d = 0.00000000000000e+00
TCL: E(abf1) = 0
TCL: E(harmonic1) = 18000
TCL: Original numsteps 20 will be ignored.
TCL: Running for 20 steps
TCL: d = 1.28684277602119e+01
TCL: E(abf1) = 0
TCL: E(harmonic1) = 94.2708
TCL: Running cv bias harm2 getconfig
TCL: 
TCL: Running for 20 steps
TCL: Running cv colvar d state
TCL: 
TCL: Running cv bias harmonic1 state
TCL: 
TCL: Running cv bias harmonic1 delete
TCL: 
TCL: Running cv bias harm2 energy
TCL: 17.8166
TCL: Running cv bias harm2 update
TCL: 17.8166
TCL: Running for 20 steps
TCL: Running cv bias harm2 energy
TCL: 13.4817
TCL: Running cv bias harm2 state
TCL: 
TCL: Running cv bias harm2 delete
TCL: 
TCL: Summary for collective variable d:
TCL:   Type: scalar number
TCL:   Width: 2.00000000000000e-01
TCL:   Atom groups: {3 } {98 } 
TCL:   Gradients computed? (0/1): 1
TCL:   Atom IDs: 
TCL:   Gradients: 
TCL:   Applied force: 2.59631860820309e+01
TCL:   Total force: -1.36757725642476e+02
TCL: Re-saving the state with a prefix:
TCL: Re-saving the state with an explicit name:
TCL: And reload:
TCL: Running cv colvar d delete
TCL: 
